src/charm.py

```diff
@@ -4,101 +4,173 @@
+        )
+        # Add initial Pebble config layer using the Pebble API
+        container = self.unit.get_container(self.name)
+        plan = container.get_plan()
+
+        if plan.services != pebble_layer["services"]:
+            logger.info("Config changed")
+            container.add_layer(self.name, pebble_layer, combine=True)
+
+            if container.get_service(self.name).is_running():
+                container.stop(self.name)
+
+            container.start(self.name)
+            logging.info("mysqlrouter restarted")
+
+        self.unit.status = ActiveStatus()
 
-    def _on_httpbin_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
+    def _on_mysqlrouter_pebble_ready(self, event) -> None:
+        """Define and start a workload using the Pebble API."""
+        if not self.has_db:
+            logger.debug("No database relation found")
+            self.unit.status = WaitingStatus("Waiting for database relation")
+            event.defer()
+            return
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
+        self._configure()
 
-        Learn more about Pebble layers at https://github.com/canonical/pebble
+    def _mysqlrouter_layer(self, host, port, user, password) -> dict:
+        """Return a layer configuration for the mysqlrouter service.
+
+        Args:
+            host (str): The hostname of the MySQL cluster.
+            port (int): The port of the MySQL cluster.
+            user (str): The username for the MySQL cluster.
+            password (str): The password for the MySQL cluster.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
+        return {
+            "summary": "mysqlrouter layer",
+            "description": "pebble config layer for mysqlrouter",
             "services": {
-                "httpbin": {
+                "mysqlrouter": {
                     "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
+                    "summary": "mysqlrouter",
+                    "command": "./run.sh",
                     "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
+                    "environment": {
+                        "MYSQL_PORT": port,
+                        "MYSQL_HOST": host,
+                        "MYSQL_USER": user,
+                        "MYSQL_PASSWORD": password,
```

Comment:
This can be me not understanding how the config file is being handled, but would it make sense to make this variables part of the class as config/self attributes or something like that?

Reply:
This will change in the future. Right now, this is just the minimal envvars for the upstream oci image to work. We will have or own image shortly.